File: contenidos/_static/code/oop-java/Estudiante.java
public class Estudiante extends Persona {
    private int legajo;

    public Estudiante(String nombre, String direccion, String dni, int legajo) {
        super(nombre, direccion, dni);
        this.legajo = legajo;
    }

    @Override
    public String saludar() {
        return "¡Hola! Soy " + getNombre() + " y soy estudiante (legajo: " + legajo + ")";
    }

    public String cursarMateria(String materia) {
        return getNombre() + " se inscribió en " + materia;
    }
}

File: contenidos/_static/code/oop/persona/estudiante.go
package persona

import "fmt"

type Estudiante struct {
	Persona
	legajo int
}

func NewEstudiante(nombre, direccion, dni string, legajo int) (*Estudiante, error) {
	if legajo <= 0 {
		return nil, fmt.Errorf("el legajo debe ser positivo, se recibió: %d", legajo)
	}
	p, err := NewPersona(nombre, direccion, dni)
	if err != nil {
		return nil, err
	}
	return &Estudiante{Persona: *p, legajo: legajo}, nil
}

func (e Estudiante) Saludar() string {
	return fmt.Sprintf("¡Hola! Soy %s y soy estudiante (legajo: %d)", e.nombre, e.legajo)
}

func (e Estudiante) CursarMateria(materia string) string {
	return fmt.Sprintf("%s se inscribió en %s", e.nombre, materia)
}
